controllers/nodehealthcheck_controller.go

```diff
@@ -42,6 +42,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/utils/pointer"
+	"k8s.io/utils/ptr"
```

Comment:
`"k8s.io/utils/pointer"` is deprecated in favor of `"k8s.io/utils/ptr"`, so we can introduce the needed changes here and remove importing module `"k8s.io/utils/pointer"` 
>Deprecated: Use functions in k8s.io/utils/ptr instead: ptr.To to obtain a pointer, ptr.Deref to dereference a pointer, ptr.Equal to compare dereferenced pointers.